fms2_io/fms2_io.F90

```diff
@@ -210,5 +211,28 @@ module fms2_io_mod
   module procedure restore_domain_state_wrap
 end interface read_new_restart
 
+!< Namelist variables
+integer :: ncchksz = 64*1024  !< User defined chunksize (in bytes) argument in netcdf file 
+                              !! creation calls. Replaces setting the NC_CHKSZ environment variable.
+namelist / fms2_io_nml / &
+                      ncchksz
+
+contains
+
+!> @brief Reads the fms2_io_nml
+subroutine fms2_io_init ()
+
+integer :: mystat
+!> Read the namelist
+READ (input_nml_file, NML=fms2_io_nml, IOSTAT=mystat)
+
+!>Send the namelist variables to their respective modules
+  if (ncchksz .le. 0) then
+        call mpp_error(FATAL, "ncchksz in fms2_io_nml must be a positive number.")
+  endif
```

Comment:
Normally, we don't assume everything we need has been initialized.  I recommend calling mpp_init, mpp_domains_init, and anything other prerequisites from FMS that fms2_io needs.